datajoint/user_relations.py

```diff
@@ -2,59 +2,83 @@
 Hosts the table tiers, user relations should be derived from.
 """
 
-import abc
 from .base_relation import BaseRelation
 from .autopopulate import AutoPopulate
 from .utils import from_camel_case
 from . import DataJointError
 
 
-class Part(BaseRelation, metaclass=abc.ABCMeta):
+class classproperty:
```

Comment:
Last time I brought up the case for stuff like `table_name` as I felt many of these values do not depend on the instance but still needs to be computed. We avoided using `classproperty` ultimately and ended up converting all of such cases into instance level property, requiring us to instantiate the class to access some of these properties. This probably means that we can rewrite the `schema` to take advantage of class property.

Anyway, I'm just giving you a bit of hard time, and I agree that use of `classproperty` is nice (after all I pushed for it last time). I just felt compelled to comment on this :P